src/assets.zig

```diff
@@ -402,30 +456,46 @@ pub fn loadWorldAssets(assetFolder: []const u8, blockPalette: *Palette, biomePal
 		_ = main.models.registerModel(entry.key_ptr.*, entry.value_ptr.*);
 	}
 
+	// items #1
+	for(itemPalette.palette.items) |id| {
+		// Normal items should appear in items hash map.
+		if(items.get(id)) |zon| {
+			try registerItem(assetFolder, id, zon);
+			continue;
+		}
+		// But there are also items created automatically from blocks.
+		if(blocks.get(id)) |zon| {
+			if(!zon.get(bool, "hasItem", true)) continue;
+			try registerItem(assetFolder, id, zon.getChild("item"));
+			continue;
+		}
+		std.log.err("Missing item: {s}. Replacing it with default item.", .{id});
+		try registerItem(assetFolder, id, .null);
+	}
+
```

Comment:
Is there still a reason why the first items need to be registered before blocks?

Reply:
Not anymore I think

Reply:
Pulled some tricks to ensure that.